GunShipAlpha/src/Enemy.java

```diff
@@ -1,41 +1,94 @@
 import java.awt.*;
+import java.awt.geom.Point2D;
+import java.util.Random;
+import java.util.Timer;
+import java.util.TimerTask;
 
 class Enemy {
-    private int x, y;
-    private final int targetX, targetY;
+    private Point location;
+    private Point target;
     private int health = 2;
+    private Timer firingTimer;
+
     public Enemy(int x, int y, int targetX, int targetY) {
-        this.x = x;
-        this.y = y;
-        this.targetX = targetX;
-        this.targetY = targetY;
+        this.location = new Point(x, y);
```

Comment:
i'd probably have taken points as the arguments instead of passing them in. seems cleaner.